core/__tests__/actions/appRefreshQuery.ts

```diff
@@ -0,0 +1,195 @@
+      test("an administrator can list all the appRefreshQueries", async () => {
+        connection.params = { csrfToken };
+        const { error, appRefreshQueries, total } =
+          await specHelper.runAction<AppRefreshQueriesList>(
+            "appRefreshQueries:list",
+            connection
+          );
+        expect(error).toBeUndefined();
+        expect(appRefreshQueries.length).toBe(1);
+        expect(appRefreshQueries[0].refreshQuery).toBe("SELECT 'hi' AS name");
+      });
+      test("an administrator can edit an appRefreshQuery", async () => {
+        connection.params = {
+          csrfToken,
+          id: id,
+          refreshQuery: "SELECT 'hello' AS name",
+        };
+        const { error, appRefreshQuery } =
+          await specHelper.runAction<AppRefreshQueryEdit>(
+            "appRefreshQuery:edit",
+            connection
+          );
+        expect(error).toBeUndefined();
+        expect(appRefreshQuery.id).toBeTruthy();
+        expect(appRefreshQuery.refreshQuery).toBe("SELECT 'hello' AS name");
+        expect(appRefreshQuery.state).toBe("draft");
+        expect(configSpy).toBeCalledTimes(1);
+      });
+      test("an administrator can make an appRefreshQuery ready", async () => {
+        connection.params = {
+          csrfToken,
+          id,
+          state: "ready",
+        };
+        const { error, appRefreshQuery } =
+          await specHelper.runAction<AppRefreshQueryEdit>(
+            "appRefreshQuery:edit",
+            connection
+          );
+        expect(error).toBeUndefined();
+        expect(appRefreshQuery.state).toBe("ready");
+      });
+      test("an administrator can test an appRefreshQuery", async () => {
+        connection.params = {
+          csrfToken,
+          id,
+        };
+        const { error, test, appRefreshQuery } =
+          await specHelper.runAction<AppRefreshQueryTest>(
+            "appRefreshQuery:test",
+            connection
+          );
+        expect(error).toBeUndefined();
+        expect(test.success).toBeTruthy();
+        expect(test.message.length).toBe(13); //test plugin queries return a unix timestamp
+        expect(appRefreshQuery.value).toBeFalsy(); //tests shouldn't save any values
+      });
+      test("an administrator can query with an appRefreshQuery", async () => {
+        connection.params = {
+          csrfToken,
+          id,
+        };
+        const { error, appRefreshQuery, valueUpdated } =
+          await specHelper.runAction<AppRefreshQueryQuery>(
+            "appRefreshQuery:query",
+            connection
+          );
+        expect(error).toBeUndefined();
+        expect(appRefreshQuery.id).toBeTruthy();
+        expect(appRefreshQuery.refreshQuery).toBe("SELECT 'hello' AS name");
+        expect(appRefreshQuery.value.length).toBe(13);
+        expect(valueUpdated).toBeTruthy();
+      });
+      test("querying with an appRefreshQuery triggers associated schedule runs", async () => {
```

Comment:
If it seems valuable to have a reverse test here (that schedules are not run if the value is the same as before), I'll need to update what the test-plugin does for its `pluginMethods.query()`... it currently sends back a NOW() timestamp, so it is always different.

Reply:
I think that logic is already tested in the model tests